apps/dashboard/src/components/workflow-editor/steps/use-ai-suggestions.tsx

```diff
@@ -0,0 +1,235 @@
+
+  const handleAiInsert = useCallback(
+    (content: GenerateContentResponse['content'], suggestedPayload?: Record<string, string>) => {
+      // Insert content into form
+      switch (step.type) {
+        case StepTypeEnum.EMAIL:
+          if ('subject' in content && 'body' in content) {
+            const emailContent = content as EmailContent;
+            const bodyStr = typeof emailContent.body === 'object' ? JSON.stringify(emailContent.body) : emailContent.body;
+
+            form.setValue('body', '');
+
+            requestAnimationFrame(() => {
+              requestAnimationFrame(() => {
+                form.setValue('subject', emailContent.subject);
+                form.setValue('body', bodyStr);
+                form.trigger(['subject', 'body']);
+              });
+            });
+          }
+          break;
+        case StepTypeEnum.SMS:
+        case StepTypeEnum.CHAT:
+          if ('body' in content) {
+            form.setValue('body', (content as SmsContent | ChatContent).body, { shouldDirty: true, shouldTouch: true });
+          }
+          break;
+        case StepTypeEnum.PUSH:
+          if ('subject' in content && 'body' in content) {
+            const pushContent = content as PushContent;
+            form.setValue('subject', pushContent.subject, { shouldDirty: true, shouldTouch: true });
+            form.setValue('body', pushContent.body, { shouldDirty: true, shouldTouch: true });
+          }
+          break;
+        case StepTypeEnum.IN_APP:
+          if ('body' in content) {
+            const inAppContent = content as InAppContent;
+            if (inAppContent.subject) {
+              form.setValue('subject', inAppContent.subject, { shouldDirty: true, shouldTouch: true });
+            }
+            form.setValue('body', inAppContent.body, { shouldDirty: true, shouldTouch: true });
+
+            if (inAppContent.primaryAction) {
+              form.setValue('primaryAction.label', inAppContent.primaryAction.label, {
+                shouldDirty: true,
+                shouldTouch: true,
+              });
+              if (inAppContent.primaryAction.url) {
+                form.setValue('primaryAction.redirect.url', inAppContent.primaryAction.url, {
+                  shouldDirty: true,
+                  shouldTouch: true,
+                });
+              }
+            }
+
+            if (inAppContent.secondaryAction) {
+              form.setValue('secondaryAction.label', inAppContent.secondaryAction.label, {
+                shouldDirty: true,
+                shouldTouch: true,
+              });
+              if (inAppContent.secondaryAction.url) {
+                form.setValue('secondaryAction.redirect.url', inAppContent.secondaryAction.url, {
+                  shouldDirty: true,
+                  shouldTouch: true,
+                });
+              }
+            }
+          }
+          break;
+      }
+
+      // Merge suggested payload into preview sandbox
+      try {
```

Comment:
Something is not working here:

https://github.com/user-attachments/assets/b3d5c61e-bf5c-439b-b9d8-2952958471db

Reply:
This is fixed